File: app/src/main/java/com/lttrung/dormitory/ui/viewrooms/ViewRoomsUseCase.java
package com.lttrung.dormitory.ui.viewrooms;

import java.lang.reflect.Type;

import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Single;

@Singleton
public interface ViewRoomsUseCase {
    // Bổ sung params, thay đổi Type bằng class tương ứng
    Single<Type> getRooms();
}

File: app/src/main/java/com/lttrung/dormitory/ui/viewrooms/ViewRoomsUseCase.kt
package com.lttrung.dormitory.ui.viewrooms

import com.lttrung.dormitory.database.data.network.models.Room
import io.reactivex.rxjava3.core.Single
import java.lang.reflect.Type
import javax.inject.Singleton

@Singleton
interface ViewRoomsUseCase {
    fun getRooms(roomTypeId: Int): Single<List<Room>>
}